pulp_rpm/tests/functional/api/test_publish.py

```diff
@@ -72,6 +72,45 @@ def test_publish_any_repo_version(
             }
             rpm_publication_api.create(body)
 
+    @pytest.mark.parametrize("compression_type", ("gz", "zstd"))
+    @pytest.mark.parallel
+    def test_publish_with_compression_types(
+        self,
+        compression_type,
+        rpm_unsigned_repo_immediate,
+        rpm_publication_api,
+        gen_object_with_cleanup,
+        rpm_distribution_api,
+        monitor_task,
+    ):
+        """Sync and publish an RPM repository w/ zstd compression and verify it exists."""
+        # 1. Publish and distribute
+        publish_data = RpmRpmPublication(
+            repository=rpm_unsigned_repo_immediate.pulp_href, compression_type="zstd"
+        )
+        publish_response = rpm_publication_api.create(publish_data)
+        created_resources = monitor_task(publish_response.task).created_resources
+        publication_href = created_resources[0]
+
+        body = gen_distribution(publication=publication_href)
+        distribution = gen_object_with_cleanup(rpm_distribution_api, body)
+
+        # 2. check "primary", "filelists", "other" and "updatedinfo" are in .zstd format
+        repomd = ElementTree.fromstring(
+            requests.get(os.path.join(distribution.base_url, "repodata/repomd.xml")).text
+        )
+        xpath_data = "{{{}}}data".format(RPM_NAMESPACES["metadata/repo"])
+        xpath_location = "{{{}}}location".format(RPM_NAMESPACES["metadata/repo"])
+
+        metadata_elements = (
+            elem
+            for elem in repomd.findall(xpath_data)
+            if elem.get("type") in ("primary", "filelists", "other", "updateinfo")
+        )
+        for elem in metadata_elements:
+            elem_href = elem.find(xpath_location).get("href")
+            assert elem_href.endswith(".zst")
```

Comment:
Likewise the test is only checking for a `.zst` extension regardless of the value of `compression_type`